djangocms_alias/models.py

```diff
@@ -127,7 +127,18 @@ def objects_using(self):
         return list(objects)
 
     def get_name(self, language=None):
-        return getattr(self.get_content(language), 'name', 'Alias {} (No content)'.format(self.pk))
+        content = self.get_content(language, access_versioning=True)
+        name = getattr(content, 'name', 'Alias {} (No content)'.format(self.pk))
+        draft = False
+        if is_versioning_enabled() and content:
+            from djangocms_versioning.constants import DRAFT
+            version = content.versions.first()
```

Comment:
This here incur extra queries. It might be a good idea to prefetch these generic relations. Pre-fetching them is a bit tricky. This method has one approach to pre-fetching generic foreign keys to reduce queries: https://github.com/v1k45/django-notify-x/pull/19/files#diff-46948b6799b46d06f97d270896f51a76R20